src/main/webapp/city.js

```diff
@@ -0,0 +1,96 @@
+/**
+ * @param {query} id
+ */
+async function searching(id) {
+  var cityId = null;
+
+  const citiesResponse = await fetch("/api/city.html?id=" + id);
+  const cities = await citiesResponse.json();
+
+  for (let city in cities) {
+    if ((cities[city]["id"] = id)) {
+      cityId = cities[city]["id"];
+    }
+  }
```

Comment:
Per the [API docs](https://stoplight.io/p/docs/gh/gferioli0418/travelbud/reference/Travelbud.v1.yaml/paths/~1api~1cities~1%7BcityID%7D/get?srn=gh/gferioli0418/travelbud/reference/Travelbud.v1.yaml/paths/~1api~1cities~1{cityID}/get&group=api-schema) the actual path to get the object for a specific city is `/api/cities/{cityID}` so this is as simple as:
```suggestion
  const cityResponse = await fetch("/api/cities/" + id);
  const city = await cityResponse.json();
```

Reply:
thanks this makes a lot more sense now.